hkmc2/shared/src/main/scala/hkmc2/codegen/Lowering.scala

```diff
@@ -729,3 +731,21 @@ trait LoweringTraceLog
     )
 
 
+object MergeMatchArmTransformer extends BlockTransformer(new SymbolSubst()):
+  override def applyBlock(b: Block): Block = b match
```

Comment:
Instead of all the boilerplate in this case, why don't you rewrite this match to
```suggestion
  override def applyBlock(b: Block): Block = super.applyBlock(b) match
```
so that you don't need to manually apply the transformation to all the extracted components yourself?

Reply:
Thanks for the suggestion! Indeed this is much cleaner, I will make the change.